src/pe/data_directories.rs

```diff
@@ -18,6 +18,25 @@ impl DataDirectory {
     pub fn parse(bytes: &[u8], offset: &mut usize) -> error::Result<Self> {
         Ok(bytes.gread_with(offset, scroll::LE)?)
     }
+
+    /// Given a view of the PE binary, represented by `bytes` and the on-disk offset `disk_offset`
+    /// this will return a view of the data directory's contents.
+    /// If the range are out of bands, this will fail with a [`error::Error::Malformed`] error.
+    pub fn data<'a>(&self, bytes: &'a [u8], disk_offset: usize) -> error::Result<&'a [u8]> {
```

Comment:
I wonder if this should have just returned an `Option` ? sort of like `get()` for arrays; e.g., is the error message actually useful? there's only one parameter passed in, so similar to an index, it means it's the wrong value basically (bad offset). anyway it's already here so it's probably fine, other than it having to allocate for the format.